edb/repl/__init__.py

```diff
@@ -435,6 +435,151 @@ def _list_scalar_types(
+                    ],
+                    data=result,
+                    max_width=min(max_width, 120),
+                )
+
+            else:
+                print(f'No object types found matching '
+                      f'{eql_quote.quote_literal(pattern)}')
+
+    @_command('lc', R'\lc [PATTERN]', 'list casts')
+    def command_list_casts(self, args: str, *,
+                           case_sensitive: bool = False) -> None:
+        self._list_casts(args.strip(), case_sensitive=case_sensitive)
+
+    @_command('lcI', R'\lcI [PATTERN]',
+              'list casts (case sensitive pattern match)')
+    def command_list_casts_i(self, args: str) -> None:
+        self.command_list_casts(args, case_sensitive=True)
+
+    def _list_casts(
+        self,
+        pattern: str,
+        case_sensitive: bool = False
+    ) -> None:
+        if case_sensitive:
+            flag = ''
+        else:
+            flag = 'i'
+
+        filter_clause, qkw = utils.get_filter_based_on_pattern(
+            pattern, ['.from_type_name', '.to_type_name'], flag=flag)
+
+        base_query = r'''
+            WITH MODULE schema
+            SELECT Cast {
+                from_type_name := .from_type.name,
+                to_type_name := .to_type.name,
+                kind := (
+                    'implicit' IF .allow_implicit ELSE
+                    'assignemnt' IF .allow_assignment ELSE
+                    'regular'
+                ),
+                volatility,
+            }
+        '''
+
+        try:
+            result, _ = self.fetch(
+                f'''
+                    {base_query}
+                    {filter_clause}
+                    ORDER BY .from_type.name THEN .to_type.name;
+                ''',
+                json=False,
+                **qkw
+            )
+        except edgedb.EdgeDBError as exc:
+            render.render_exception(self.context, exc)
+        else:
+            if result:
+                print(f'List of casts:')
+                max_width = self.prompt.output.get_size().columns
+                render.render_table(
+                    self.context,
+                    title='Casts',
+                    columns=[
+                        table.ColumnSpec(
+                            field='from_type_name', title='From Type',
+                            width=1, align='left'),
+                        table.ColumnSpec(
+                            field='to_type_name', title='To Type',
+                            width=1, align='left'),
+                        table.ColumnSpec(
+                            field='kind', title='Type of Cast',
+                            width=1, align='left'),
+                        table.ColumnSpec(
+                            field='volatility', title='Volatility',
+                            width=1, align='left'),
+                    ],
+                    data=result,
+                    max_width=min(max_width, 120),
```

Comment:
Please move 120 into a constant (MAX_TABLE_WIDTH or something)

Reply:
It's also used for other output formatting, so I'll just call it `MAX_WIDTH` and add a short comment.